storm-core/src/clj/org/apache/storm/daemon/executor.clj

```diff
@@ -267,9 +267,11 @@
                                ((:report-error <>) error)
                                (catch Exception e
                                  (log-message "Error while reporting error to cluster, proceeding with shutdown")))
-                             (if (or
-                                    (exception-cause? InterruptedException error)
-                                    (exception-cause? java.io.InterruptedIOException error))
+                             (if (and
```

Comment:
https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/executor/error/ReportErrorAndDie.java#L41-L43

According to current implementation for master branch, it should be

```
(if (or
  (exception-cause? InterruptedException error)
  (and
    (exception-cause? java.io.InterruptedIOException error)
    (not (exception-cause? java.net.SocketTimeoutException error))))))
```